scala/private/macros/scala_repositories.bzl

```diff
@@ -213,13 +236,19 @@ def scala_repositories(
         overriden_artifacts = {},
         load_dep_rules = True,
         load_jar_deps = True,
-        fetch_sources = False):
+        fetch_sources = False,
+        validate_scala_version = True,
+        scala_compiler_srcjars = {}):
     if load_dep_rules:
-        rules_scala_setup()
+        # When `WORKSPACE` goes away, so can this case.
+        rules_scala_setup(setup_compiler_sources = False)
```

Comment:
I'd agree, but that would be another break from the existing API. But if you definitely want it now, I'll do it.

Reply:
Just to make myself clear. What I meant was
```
def rules_scala_setup(scala_compiler_srcjar = None):
  load_rules_dependencies()
  setup_scala_compiler_sources()
```
So that existing clients of `rules_scala_setup` won't break

And here have 
```
if load_dep_rules:
   load_rules_dependencies()
setup_scala_compiler_sources(scala_compiler_srcjars)
```

This would add two new macros to the api but it would be easier to follow the code (at least for me). Unless I completely misunderstood what you mean by "break from the existing API"

Reply:
Ah, you're so right. I wasn't reading closely enough or thinking clearly enough before. This is definitely better, and doesn't break the existing API at all. Done.